### PR TITLE
Add to development process FT and Feature request

### LINKED ISSUE #52: [Functional Tests] Document about the functional tests
Add to the `docs/source/devel` the documentation about the Functional Tests.

Some topics that will be better explained in the documentation
- The functional tests will cover Features by Behavior
   - User interactions between component
      Example: "Metadata Retention". You don't want to test the Metadata Retention, but the API user can change the Metadata Retention value.
- TRS will use Behavior Driven Development (BDD) test
- The BDD tests will be used for Feature proposals/changes
   - Providing the [gherkin](https://cucumber.io/docs/gherkin/reference/) in a [.feature](https://pytest-bdd.readthedocs.io/en/latest/#example) file will be required

### PR DESCRIPTION
**Pull request by kairoaraujo**
This commits add the documentation about the Functional Tests (BDD) and the process for Feature request/proposal.

Also add simple explanation about general UT and toolings.

Closes #52

Signed-off-by: Kairo de Araujo <kdearaujo@vmware.com>

### CODE REVIEW

File: docs/source/devel/development.rst [hunk 1]
```diff
@@ -136,15 +131,111 @@ Working on the Component Issues:
...
+
+Component-only feature
+----------------------
+
+The component-only features are related and internal to a component (i.e.,
+Repository Service for TUF API or Repository Service for TUF Worker). These
+features don’t impact the entire RSTUF platform, such as performance,
+breaking/improving API contracts, user usage/deployment, etc.
+
+We recommend for this type of feature open a new issue in the :ref:`specific
```
Comment (rdimitrov) [inline docs/source/devel/development.rst:147]:
```suggestion
We recommend for this type of feature to open a new issue in the :ref:`specific
```

File: docs/source/devel/development.rst [hunk 2]
```diff
@@ -136,15 +131,111 @@ Working on the Component Issues:
...
+We recommend for these changes the following:
+
+* Always design/document the feature in a `new issue
+  <https://github.com/vmware/repository-service-tuf/issues/new?assignees=&labels=feature&template=feature.yml&title=Feature%3A+>`_.
+
+  - Use as many diagrams as possible (PlantUML or Mermaid in the issues). ❤️
+
+* After discussing and receiving enough feedback, propose a :ref:`BDD Feature file
+  <devel/development:Functional Tests using Behavior Development Driven (BDD)>`
+  (ask for a feature brach).
```
Comment (rdimitrov) [inline docs/source/devel/development.rst:166]:
```suggestion
  (ask for a feature branch).
```

File: docs/source/devel/development.rst [hunk 3]
```diff
@@ -136,15 +131,111 @@ Working on the Component Issues:
...
+Tooling
+.......
+
+We use `pytest <https://docs.pytest.org/>`_ and
+`pytest-bdd <https://pytest-bdd.readthedocs.io/en/stable/>`_ as the framework.
+
+All BDD tests are in the `Umbrella repository, inside
+tests/functional <http://github.com/vmware/repository-service-tuf>`_ .
+
+The BDD tests has the workflow also in the `Umbrella repository
```
Comment (rdimitrov) [inline docs/source/devel/development.rst:222]:
```suggestion
The BDD tests have the workflow also in the `Umbrella repository
```

File: docs/source/devel/development.rst [hunk 4]
```diff
@@ -136,15 +131,111 @@ Working on the Component Issues:
...
+<http://github.com/vmware/repository-service-tuf>`_ , as a reusable
+Github Workflow, and it is triggered by other worflows/components i.e.,
```
Comment (rdimitrov) [inline docs/source/devel/development.rst:224]:
```suggestion
GitHub Workflow, and it is triggered by other workflows/components i.e.,
```

File: docs/source/devel/development.rst [hunk 5]
```diff
@@ -136,15 +131,111 @@ Working on the Component Issues:
...
+before releasing.
+
 
 Project organization
 ====================
 
-The Components have the development instructions.
+The project uses the microservices approach.
+Each RSTUF components have the own development instructions.
```
Comment (rdimitrov) [inline docs/source/devel/development.rst:232]:
```suggestion
Each RSTUF components have its own development instructions.
```

**Review by MVrachev — COMMENTED** [review]
Good additions to our documentation.

Do we want to also add:
- a specific command on how to initially generate a functional test file
- a specific command on how to run the functional test file

File: docs/source/devel/development.rst [hunk 6]
```diff
@@ -136,15 +131,111 @@ Working on the Component Issues:
...
+pytest and `pretend <https://github.com/alex/pretend>`_ to implement the
+stubbing technic.
+
+More details will be in each :ref:`specific
+component <devel/development:Components Repositories>`.
+
+Functional Tests using Behavior Development Driven (BDD)
+--------------------------------------------------------
+
+DBB Feature file
```
[the remark this answers is not in the document]
Comment (kairoaraujo) [reply docs/source/devel/development.rst:194]: Oh yes, I typed it wrong :)

**Comment by kairoaraujo** [comment]
> Good additions to our documentation.
> 
> Do we want to also add:
> 
> * a specific command on how to initially generate a functional test file
> * a specific command on how to run the functional test file

I think it is an excellent new issue when we create a specific command that generates the tests locally.
We don't support it "officially". Besides, we know that we follow the steps from the GH Workflows for functional tests.

I suggest adding it together with the smart `make functional-tests` that can identify whether it is running in the local/developer machine or the GH environment.

**Comment by MVrachev** [comment]
I still think it's better to leave a sentence or two on how one can start writing a functional test file.
At least this command could be mentioned:
`pytest-bdd generate  <PATH_TO_FEATURE_FILE>`
It will be a bonus if we mention how one could run a specific functional test file like this:
`pytest <FEATURE_FILE_PATH> --gherkin-terminal-reporter -vvv`
but at least we have `make functional-tests`.

**Comment by MVrachev** [comment]
Yes, you are right that `pytest-bdd` can be replaced by another BDD tool.
Well, the important part that we specify that we are using `pytest-bdd` is said, so from then on one can figure out how to use the tool with their documentation. 

Thanks for creating an issue for improving the `make functional-tests`.

**Review by MVrachev — APPROVED** [review]

**Pull request merged**